src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,139 @@
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import TodoTitleField from '../TodoTitleField/TodoTitleFiled';
+
+type Props = {
+  todo: Todo;
+  onDeleteTodo: (todoId: number) => void;
+  deletingCompleted: boolean;
+  onToggleStatus: () => void;
+  isChanging: boolean;
+  currentlyChanging: string;
+  onTitleChange: (todo: Todo, value: string) => void;
+};
+
+const TodoComponent: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  deletingCompleted,
+  onToggleStatus,
+  isChanging,
+  currentlyChanging,
+  onTitleChange,
+}) => {
+  const [hasSpinner, setHasSpinner] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+
+  let isActive = todo.completed && deletingCompleted ? true : hasSpinner;
+
+  if (currentlyChanging) {
+    if (todo.completed && currentlyChanging === 'completed') {
+      isActive = true;
+    }
+
+    if (!todo.completed && currentlyChanging === 'active') {
+      isActive = true;
+    }
+  }
+
+  useEffect(() => {
+    if (!isChanging) {
+      setHasSpinner(false);
+    }
+  }, [isChanging]);
+
+  const deleteHandler = (todoId:number) => {
+    onDeleteTodo(todoId);
+    setHasSpinner(true);
```

Comment:
well then I would need to set this individually to the specific todo that i am deleting, since isLoading triggers all the todos, not the individual one. That is why i need this setSpinner state that I cal only when the delete handler is called in this particular todo.

Reply:
ok